src/arch/forte_stringFunctions.h

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Samator Indo Gas
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Ketut Kumajaya - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+#ifndef SRC_ARCH_FORTE_STRINGFUNCTIONS_H_
+#define SRC_ARCH_FORTE_STRINGFUNCTIONS_H_
+
+#include <string>
+#include <codecvt>
+
+inline const std::wstring forte_stringToWstring(const std::string &paVal) {
+  auto &converter = std::use_facet<std::codecvt<char, wchar_t, std::mbstate_t>>(std::locale());
+  std::mbstate_t mb{};
+  std::wstring dst(paVal.size() * converter.max_length(), L'\0');
+  const char *from_next;
+  wchar_t *to_next;
+  converter.out(mb, &paVal[0], &paVal[paVal.size()], from_next,
+                    &dst[0], &dst[dst.size()], to_next);
+  // error checking skipped for brevity
+  dst.resize(to_next - &dst[0]);
+  return dst;
+}
+
+inline const std::string forte_wstringToString(const std::wstring &paVal) {
+  auto &converter = std::use_facet<std::codecvt<wchar_t, char, std::mbstate_t>>(std::locale());
+  std::mbstate_t mb{};
+  std::string dst(paVal.size() * converter.max_length(), '\0');
+  const wchar_t *from_next;
+  char *to_next;
+  converter.out(mb, &paVal[0], &paVal[paVal.size()], from_next,
+                    &dst[0], &dst[dst.size()], to_next);
```

Comment:
I am not sure if passing the address of the wstring's index operator is correct here, as, as far as I know, there is no guarantee that two calls are pointing to a consecutive memory of glyphs.
From my point of view, you need to call data() on the wstring to get this consecutive memory.

I would also ask you to provide some test cases for the new functions, so in case of code changes automatic tests are in place already.